Lib/typing.py

```diff
@@ -989,13 +996,21 @@ def _evaluate(self, globalns, localns, recursive_guard):
                     sys.modules.get(self.__forward_module__, None), '__dict__', globalns
                 )
             type_ = _type_check(
-                eval(self.__forward_code__, globalns, localns),
+                eval(
+                    self.__forward_code__,
+                    globalns,
+                    {param.__name__: param for param in type_params} | localns
```

Comment:
this means that any annotations in the "local namespace" of the class override any symbols that are present in the scope due to being listed as a type parameter. That matches the behaviour of `__annotations__` at runtime:

```pycon
>>> class B[T, *Ts, **P]:
...     T = int
...     Ts = str
...     P = bytes
...     x: T
...     y: Ts
...     z: P
... 
>>> B.__annotations__
{'x': <class 'int'>, 'y': <class 'str'>, 'z': <class 'bytes'>}
```

I'm doing it at the last possible point here as it feels least risky than doing it in `_eval_type`.